admin/lib/solidus_admin/preview.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+# This module will add all the necessary harness to a ViewComponent::Preview
+# to be able to render Solidus components.
+#
+# Adds a `current_component` helper method that will return the the component class
+# based on the preview class name. The component class name is inferred by removing
+# the `Preview` suffix from the preview class name.
+#
+# Adds a `helpers` method that will return the helpers module from SolidusAdmin::BaseController
+# making them available to the preview class (which is not a standard controller). All helpers
+# are available to the preview class via `method_missing`.
+#
+# @example
+#   class SolidusAdmin::UI::Badge::ComponentPreview < ViewComponent::Preview
+#     include SolidusAdmin::Preview
+#
+#     def default
+#       render current_component.new(name: time_ago_in_words(1.day.ago))
+#     end
+#   end
+#
+module SolidusAdmin::Preview
+  extend ActiveSupport::Concern
+
+  module ControllerHelper
+    extend ActiveSupport::Concern
+
+    included do
+      helper ActionView::Helpers
+      helper_method :current_component
+    end
+
+    private
+
+    def current_component
+      @current_component ||= begin
+        # Lookbook sets the @preview instance variable with a PreviewEntity instance, while ViewComponent uses the preview class.
+        # Lookbook's PreviewEntity has `#preview_class_name` as part of its public api.
+        # Since we want to support both ways or rendering components, we need to check.
+        preview_class_name = @preview.respond_to?(:preview_class_name) ? @preview.preview_class_name : @preview.name
+        preview_class_name.chomp("Preview").constantize
+      end
+    end
+  end
+
+  included do
+    layout "solidus_admin/preview"
+
+    delegate :helpers, to: SolidusAdmin::BaseController
+    private :helpers # hide it from preview "example" methods
+  end
+
+  def current_component
```

Comment:
Is there a reason not to call this method just `component`? I feel like this will be the most used interface and maybe we can save some complexity removing the `current_` prefix. 

I'm not sure about `component` either, because it matches with the `component` method name outside the preview world, but I feel like `current_` add some specificity without telling anything about why it's needed or what it does.

Thinking loud and ready to change my mind, no strong opinions here, thoughts?

Reply:
I think that's to avoid the name collision with the [`component` helper](https://github.com/solidusio/solidus/blob/64232793a1945eb80c6083e99d3d2d59c9ef3b9e/admin/app/helpers/solidus_admin/container_helper.rb#L13), which is now available to the preview classes.

Reply:
😄 I actually called it `component` initially, but soon realized what you pointed out, i.e. that we also have the `component` helper. And it's already the case that it is very useful to show components collaborating with other components for either context or to provide content.

Right now it's used to show a badge in table cells, but you can also think of a modal or a sidebar component, both needing some content to be showcased in the preview.

I picked `current_` because it seemed natural (and is very familiar) but we can switch to something like `previewed_component` or something else if you think it's worth it.

Reply:
I think I'd slightly prefer something like `previewed_component`, but I'm also ok keeping `current_component` with the above in mind. Up to you, and we can also change it later.